src/youtube_chapters_demo.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import youtube_dl
+import re
+import datetime
+
+import opentimelineio as otio
+
+
+# Input: @description_file contains the Youtube description
+# Function should parse the Youtube description to find any table-of-contents entries. 
+# The table-of-contents entries will contain a timestamp paired with a chapter title. 
+# For example: (00:12:15) Thor fights Thanos
+# 
+# The function should return an array of all such timestamps and chapter titles. 
+#
+def process_youtube_description(description_file): 
+  with open(description_file) as f:
+    lines = f.readlines()
+
+  # Should match the following timestamp patterns: 
+  # 01:34:21 chapter title
+  # 1:34:21 chapter title 
+  # 34:21 chapter title
+  # 4:21 chapter title
+  # (4:21) chapter title
+  # [4:21] chapter title
+
+
+  pattern = re.compile('((?:\d+:)+\d{2}).*\s(.+)')
+  chapters = []
+
+  for line in lines: 
+      matches = pattern.findall(line) 
+      if len(matches) > 0: 
+          chapters.append(matches[0])
+   
+  return chapters
+         
+
+
+def convert_time_stamp_to_seconds(time_stamp):
+    if len(time_stamp) > 5:
+        date_time = datetime.datetime.strptime(time_stamp, "%H:%M:%S")
+    else:
+        date_time = datetime.datetime.strptime(time_stamp, "%M:%S")
+    
+    a_timedelta = date_time - datetime.datetime(1900, 1, 1)
+    seconds = a_timedelta.total_seconds()
+    return seconds 
+
+
+def create_markers(chapters, fps):
+    markers = []
+
+    for chapter in chapters: 
+        seconds = convert_time_stamp_to_seconds(chapter[0])
+ 
+        marker = otio.schema.Marker()
+
+        marker.marked_range = otio.opentime.TimeRange(
+            start_time=otio.opentime.RationalTime(seconds*fps, fps),
+            duration=otio.opentime.RationalTime() #TODO: Set a specific duration
```

Comment:
You could consider setting the duration to 0 frames, 1 frame, or the whole span up to the next marker. Since this is meant to be sample code, it would be fine to just put a comment noting the choice without needing to make a strong statement about how it "should" be done.